### ISSUE TITLE
Error when running multiple yeoman generators simultaneously

### ISSUE THREAD

**Issue by shan916** [yeoman/yo#459]
I am writing a back-end service for a website that will run a yeoman generator and download a zip file of the generated app. I am getting errors if I run multiple instance of yo simultaneously. From my research, it appears to be an issue with multiple instances of yo trying to access the insight-yo.json file. Is this something that can be fixed with yo or is there a way I can prevent yo from gathering any type of insights? I have already tried running yo with --no-insights. I am running this on my Windows 7 machine.
Here are the errors: 

> Error: EPERM: operation not permitted, rename 'C:\Users\Zeeskh2.config\configstore\insight-yo.json.3584203431' -> 'C:\Users\Zeeskh2.config\configstore\insight-yo.json'
>     at Error (native)
>     at Object.fs.renameSync (fs.js:681:18)
>     at Function.writeFileSync [as sync](C:UsersZeeskh2AppDataRoamingnpmnode_modulesyonode_modulesconfigstorenode_moduleswrite-file-atomicindex.js:39:8)
>     at Object.create.all.set (C:\Users\Zeeskh2\AppData\Roaming\npm\node_modules\yo\node_modules\configstore\index.js:62:21)
>     at Object.Configstore (C:\Users\Zeeskh2\AppData\Roaming\npm\node_modules\yo\node_modules\configstore\index.js:27:11)
>     at new Insight (C:\Users\Zeeskh2\AppData\Roaming\npm\node_modules\yo\node_modules\insight\lib\index.js:37:34)
>     at Object.<anonymous> (C:\Users\Zeeskh2\AppData\Roaming\npm\node_modules\yo\lib\cli.js:172:11)
>     at Module._compile (module.js:409:26)
>     at Object.Module._extensions..js (module.js:416:10)
>     at Module.load (module.js:343:32)
> C:\Users\Zeeskh2\AppData\Roaming\npm\node_modules\yo\node_modules\configstore\index.js:69
>                                 throw err;
>                                 ^
> 
> Error: EPERM: operation not permitted, rename 'C:\Users\Zeeskh2.config\configstore\insight-yo.json.1025325944' -> 'C:\Users\Zeeskh2.config\configstore\insight-yo.json'
>     at Error (native)
>     at Object.fs.renameSync (fs.js:681:18)
>     at Function.writeFileSync [as sync](C:UsersZeeskh2AppDataRoamingnpmnode_modulesyonode_modulesconfigstorenode_moduleswrite-file-atomicindex.js:39:8)
>     at Object.create.all.set (C:\Users\Zeeskh2\AppData\Roaming\npm\node_modules\yo\node_modules\configstore\index.js:62:21)
>     at Object.Configstore (C:\Users\Zeeskh2\AppData\Roaming\npm\node_modules\yo\node_modules\configstore\index.js:27:11)
>     at new Insight (C:\Users\Zeeskh2\AppData\Roaming\npm\node_modules\yo\node_modules\insight\lib\index.js:37:34)
>     at Object.<anonymous> (C:\Users\Zeeskh2\AppData\Roaming\npm\node_modules\yo\lib\cli.js:172:11)
>     at Module._compile (module.js:409:26)
>     at Object.Module._extensions..js (module.js:416:10)
>     at Module.load (module.js:343:32)

**Comment by SBoudrias** [comment]
`Error: EPERM: operation not permitted` looks like a `chmod` issue rather than race condition.

**Comment by shan916** [comment]
@SBoudrias It only does that if I am running simultaneous instances of yo.

**Comment by sindresorhus** [comment]
It's a known issue: https://github.com/npm/fs-write-stream-atomic/pull/13

**Comment by sindresorhus** [comment]
You can run it with `$ yo --no-insight` to skip that step. See the flags documented in the readme.

**Comment by shan916** [comment]
@sindresorhus I still get the permission error with `--no-insight`. I just tested running multiple instances of yo in a Docker container and it doesn't get any permission errors. It does appear to be a Windows only issue with fs-write-stream-atomic as you mentioned.

**Comment by eddiemonge** [comment]
updated related issue https://github.com/npm/fs-write-stream-atomic/pull/14

**Comment by eddiemonge** [comment]
Just out of curiosity, why are you running the yo instances in parallel?

**Comment by shan916** [comment]
@eddiemonge I wrote a web app for my company that uses our custom Yeoman generator to build a starter project. 

When I performed some load testing using my Windows machine, I was getting these errors. Not that our developers are going to be generating a large number of projects in a short amount of time. Everything ran fine when I deployed to a Linux server.

I think we can go ahead and close this since the [npm/fs-write-stream-atomic#14](https://github.com/npm/fs-write-stream-atomic/pull/13) issue has been fixed.
